clang/include/clang/APINotes/Types.h

```diff
@@ -300,6 +300,72 @@ inline bool operator!=(const ContextInfo &LHS, const ContextInfo &RHS) {
   return !(LHS == RHS);
 }
 
+/* TO_UPSTREAM(BoundsSafety) ON */
+class BoundsSafetyInfo {
+public:
+  enum class BoundsSafetyKind {
```

Comment:
Doesn't this enum basically exist elsewhere? Should we be duplicating it here?

Reply:
Originally I was thinking about including `__terminated_by`, `__single`, `__bidi_indexable` etc, and I think. There's still a possibility that those may be added later, especially `__terminated_by`, but more importantly, this enum representation is now used in the binary encoding of API notes, so I think duplicating it is fine to keep them uncoupled.